### PR TITLE
1101861: retry the next link if no response found and all records method changes

### PR DESCRIPTION
**Pull request by nileshpatil-mindbody**
1101861: retry the next link if no response found and all records method changes
Ticket: [1101861](https://dev.azure.com/mindbody/Marketing%20Products/_workitems/edit/1101861)

### CODE REVIEW

**Review by moizmb — CHANGES_REQUESTED** [review]
I dont see any tests for this, can you add them please

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:latest
+FROM ruby:2.7.0
```
Comment (moizmb) [inline Dockerfile:1]: why is this required?

File: lib/frederick_api/v2/helpers/paginator.rb [hunk 2]
```diff
@@ -17,15 +17,25 @@ def all_records
...
               results.push(*current_result_set.to_a)
             end
           end
-
           results
         end
 
+        # refetching the next link if no response is found due to elastic cache unknown beheviour.
+        # bug link: https://dev.azure.com/mindbody/Marketing%20Products/_workitems/edit/1101861
+        def next_result_set(current_result_set)
```
Comment (otheogar) [inline lib/frederick_api/v2/helpers/paginator.rb:29]: I think we should make this a private method and add to the tests for all_records - add tests for when the current result set is empty.

**Review by otheogar — APPROVED** [review]

File: spec/frederick_api/v2/helpers/paginator_spec.rb [hunk 3]
```diff
@@ -77,15 +77,46 @@
       rs
     end
 
+    context 'returns result from all pages' do
+      before do
+        expect(paginator).to receive(:current_page).with(no_args).and_return(current_page)
+        expect(paginator).to receive(:total_pages).with(no_args).and_return(total_pages)
+        expect(paginator).to receive(:next).with(no_args).and_return(new_result_set)
+        expect(paginator).to receive(:next).with(no_args).and_return(new_result_set2)
```
Comment (moizmb) [inline spec/frederick_api/v2/helpers/paginator_spec.rb:85]: We dont expect in the before block, we should be expecting in an `it` block.

File: spec/frederick_api/v2/helpers/paginator_spec.rb [hunk 4]
```diff
@@ -77,15 +77,46 @@
...
     before do
-      expect(paginator).to receive(:current_page).with(no_args).and_return(current_page)
-      expect(paginator).to receive(:total_pages).with(no_args).and_return(total_pages)
       expect(paginator).to receive(:next).with(no_args).and_return(new_result_set)
-      expect(paginator).to receive(:next).with(no_args).and_return(new_result_set2)
     end
 
-    it 'aggregates all the remaining pages' do
-      expect(paginator.all_records).to eq(result + result2 + result3)
+    it 'fetch the next link' do
```
Comment (moizmb) [inline spec/frederick_api/v2/helpers/paginator_spec.rb:118]:
```suggestion
    it 'fetches the next link' do
```
